src/main/java/tech/pathtoprogramming/currencyconverter/domain/MoneyConverter.java

```diff
@@ -3,17 +3,35 @@
 import org.springframework.beans.factory.annotation.Qualifier;
 import org.springframework.stereotype.Component;
 
+import java.io.IOException;
+import java.math.BigDecimal;
+import java.math.RoundingMode;
+
 @Component
 public class MoneyConverter {
     private final ExchangeBoard exchangeBoard;
     private final MoneyCalculator moneyCalculator;
 
-    public MoneyConverter(@Qualifier("fakeExchangeBoard") ExchangeBoard exchangeBoard, MoneyCalculator moneyCalculator) {
+    public MoneyConverter(  @Qualifier("fakeExchangeBoard")
+                          ExchangeBoard exchangeBoard, MoneyCalculator moneyCalculator) {
         this.exchangeBoard = exchangeBoard;
         this.moneyCalculator = moneyCalculator;
     }
 
-    public Money convert(String basePrice, String baseCurrency, String targetCurrency) {
-        throw new UnsupportedOperationException("need to implement");
+    public Money convert(String basePrice,
+                         String baseCurrency,
+                         String targetCurrency) throws IOException {
+        Money baseMoney = Money.createMoney(basePrice, baseCurrency);
+        BigDecimal exchangeRateForBaseCurrency
+                = exchangeBoard.getExchangeRateFor(baseMoney,
+                Currency.valueOf(targetCurrency));
+
+        BigDecimal calculatedMoney
+                = moneyCalculator.calculate(BigDecimal.
+                        valueOf(Long.parseLong(basePrice)),
```

Comment:
There's a bug here. After fixing the api key issue, I ran the app and got this error for a decimal input.

`java.lang.NumberFormatException: For input string: "100.00"`

The reason for this is you are trying to convert a String - 100.00 into a Long. A Long is a whole number.
You could address this in two different ways:

- convert the string into a double instead 
- use the constructor call of BigDecimal that takes a String

```java
BigDecimal calculatedMoney = moneyCalculator.calculate(
                BigDecimal.valueOf(Double.parseDouble(basePrice)),
                exchangeRateForBaseCurrency);                        
```

OR

```java
BigDecimal calculatedMoney = moneyCalculator.calculate(
                new BigDecimal(basePrice),
                exchangeRateForBaseCurrency);
```

Reply:
Alright this has helped the issue thanks